### PR TITLE
Refactor macros

### PR DESCRIPTION
**Pull request by calebmer**

### CODE REVIEW

File: src/value.rs [hunk 1]
```diff
@@ -35,11 +43,10 @@ pub enum Value {
   F64(f64),
```
Comment (svmnotn) [inline src/value.rs:43]: I've refrained from asking for far too long but. why not `float(f64)` and `int(isize)`?

**Comment by svmnotn** [comment]
Other than that about `Value` which probably belongs in it's own PR, :+1: for merge

File: src/value.rs (hunk 1 above)
Comment (calebmer) [inline src/value.rs:43]: I've seen implementations which use both `I64` and `I32`. Basically, everyone else is doing it 😊

I think this is a large design decision that we need more people to weigh in on, so I'm going to open an issue we can collect votes/opinions in.

Also, what's the difference between `isize` and `i64`?

Comment (svmnotn) [inline src/value.rs:43]: `isize` means that compiled in a 64-bit sys it is `i64`, if compiled in a 32-bit sys it is `i32`

Comment (calebmer) [inline src/value.rs:43]: Ah, add a comment to the issue with that concern.

**Pull request merged**